### PR TITLE
design(functions/fill): add design for fill operation

### PR DESCRIPTION
**Pull request by aanthony1243**
### Done checklist
- [X] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -1218,6 +1218,21 @@ Example:
 
 **Note:** The `yield` function produces side effects.
 
+#### Fill
+
+Fill will scan a stream for nil values and replace them with a non-null value.
```
Comment (rbetts) [inline docs/SPEC.md]: s/non-null/non-nil/?

File: docs/SPEC.md [hunk 2]
```diff
@@ -1218,6 +1218,21 @@ Example:
 
 **Note:** The `yield` function produces side effects.
 
+#### Fill
+
+Fill will scan a stream for nil values and replace them with a non-null value.  
+
+The output stream will be the same as the input stream, with all null values replaced.
```
Comment (rbetts) [inline docs/SPEC.md]: nil/null

File: docs/SPEC.md [hunk 3]
```diff
@@ -1218,6 +1218,21 @@ Example:
...
+Fill will scan a stream for nil values and replace them with a non-null value.  
+
+The output stream will be the same as the input stream, with all null values replaced.  
+
+Fill has the following properties: 
+
+* `valueColumn`  string
+    the column to fill
+* `value`        bool, int, uint, float, string, time
+    the constant value to use in place of nulls
```
Comment (rbetts) [inline docs/SPEC.md]: Does the `value` type have to match the column type?

File: docs/SPEC.md (hunk 1 above)
Comment (aanthony1243) [reply docs/SPEC.md]: the rest of the spec uses `null`.  I'm ok with changing it, but it'd need to change everywhere.

Comment (aanthony1243) [reply docs/SPEC.md]: ...except for this PR, where I use one of each for every other sentence.   Fixed all instances to null

File: docs/SPEC.md [hunk 4]
```diff
@@ -1218,6 +1218,21 @@ Example:
 
 **Note:** The `yield` function produces side effects.
 
+#### Fill
+
+Fill will scan a stream for null values and replace them with a non-null value.  
+
+The output stream will be the same as the input stream, with all null values replaced.
```
Comment (nathanielc) [inline docs/SPEC.md]: `with all null values in the column replaced`

File: docs/SPEC.md [hunk 5]
```diff
@@ -1218,6 +1218,21 @@ Example:
...
 
+#### Fill
+
+Fill will scan a stream for null values and replace them with a non-null value.  
+
+The output stream will be the same as the input stream, with all null values replaced.  
+
+Fill has the following properties: 
+
+* `valueColumn`  string
```
Comment (nathanielc) [inline docs/SPEC.md]: This should just be `column` to be consistent.

**Pull request merged**